docs/docs/building-ui/composite-components.md

```diff
@@ -1,92 +1,163 @@
 
-:::tip
-It's highly recommended to create custom components by utilizing the `Composite` component, rather than extending the base `Component` component.
-:::
+A `Composite` component has a strong association with an underlying bound component. This gives you control over which methods and properties users can access, unlike traditional inheritance where everything is exposed.
+
+If you need to integrate web components from another source, use specialized alternatives:
+
+- ElementComposite: For web components with type-safe property management
+- ElementCompositeContainer: For web components that accept slotted content
 
-To utilize the `Composite` component, start by creating a new Java class that extends the `Composite` component. Specify the type of Component you want to manage as the generic type parameter.
+## Usage {#usage}
+
+To define a `Composite` component, extend the `Composite` class and specify the type of component it manages. This becomes your bound component, which is the root container that holds your internal structure. 
 
 ```java
-public class ApplicationComponent extends Composite<Div> {
-	//Implementation
+public class UserCard extends Composite<FlexLayout> {
+    // Implementation
 }
 ```
 
-## Component binding {#component-binding}
-
-The `Composite` class requires developers to specify the type of `Component` it manages. This strong association ensures that a `Composite` component is intrinsically linked to its underlying Component. This also provides benefits over traditional inheritance, as it allows the developer to decide exactly what functionality to expose to the public API. 
+The bound component can be any [webforJ component](../components/overview) or [HTML element component](/docs/building-ui/web-components/html-elements). For flexible layouts, consider using [`FlexLayout`](../components/flex-layout) or Div as your bound component.
 
-By default, the `Composite` component utilizes the generic type parameter of its subclass to identify and instantiate the bound component type. This is based on the assumption that the component class has a parameter-less constructor. Developers can customize the component initialization process by overriding the `initBoundComponent()` method. This allows for greater flexibility in creating and managing the bound component, including invoking parameterized constructors.
+:::note Component Extension
+Never extend `Component` or `DwcComponent` directly. Always use composition patterns with `Composite` to build custom components.
+:::
 
-The following snippet overrides the initBoundComponent method to use a parameterized constructor for the [FlexLayout](../components/flex-layout.md) class:
+By default, the bound component is instantiated using its no-argument constructor. To customize this behavior, override the 	`initBoundComponent()` method:
```

Comment:
The section does not explain why this is needed. Check original docs: 

https://docs.webforj.com/docs/building-ui/composite-components

Reply:
The current documentation doesn’t explain why someone would ever re-implement the initBoundComponent.
read this: https://docs.webforj.com/docs/building-ui/composite-components#component-binding